src/js/letters/containers/AddressSection.jsx

```diff
@@ -29,6 +28,12 @@ import {
 } from '../utils/validations';
 import { AVAILABILITY_STATUSES } from '../utils/constants';
 
+// The address is empty if every field except type is falsey.
+// NOTE: It "shouldn't" ever happen, but it did in testing, so...paranoid programming!
+function isAddressEmpty(address) {
+  return Object.keys(address).reduce((emptySoFar, nextField) => emptySoFar && (nextField === 'type' || !address[nextField]), true);
+}
```

Comment:
Think I'm reviewing this too late in the day because this took longer to understand than I care to admit 🐢

Reply:
Yeaaaaaaah, it's not _super_ clear, but the alternative was breaking the callback onto multiple lines. If that'll make it clearer, I'd be happy to do that. I wasn't thrilled with the length of this line. 0.o